animations/lib/src/misc/animated_list.dart

```diff
@@ -20,17 +20,18 @@ class _AnimatedListDemoState extends State<AnimatedListDemo> {
     setState(() {
       var index = listData.length;
       listData.add(
-        UserModel(firstName: 'New', lastName: 'Person'),
+        UserModel(++_maxIdValue, 'New', 'Person'),
       );
       _listKey.currentState
-          .insertItem(index, duration: Duration(milliseconds: 300));
+          ?.insertItem(index, duration: Duration(milliseconds: 300));
```

Comment:
I *think* it makes more sense to use `!`, since this would only be null if something is set up incorrectly:

```
  /// The current state is null if (1) there is no widget in the tree that
  /// matches this global key, (2) that widget is not a [StatefulWidget], or the
  /// associated [State] object is not a subtype of `T`.
```

Reply:
Done.